src/database.cpp

```diff
@@ -310,14 +310,16 @@ bool Database::load()
          }
    }
 
-   // With the db open, we can get all our hashes. This has to happen before
-   // updateSchema
-   Database::tableNames = tableNamesHash();
-   Database::classNameToTable = classNameToTableHash();
```

Comment:
It is mostly wrong, but being mostly wrong is like being mostly dead.

We have to populate the hashes before we call populateChildrenTables(), which is called by updateSchema(). The problem was that until DatabaseSchemaHelper::migrate() is called to do the v5 -> v6 upgrade the table that populates the hash doesn't exist. This causes populateChildrenTables to throw an exception.

I fixed this the first time by changing the order and populating the hashes first. That is when I wrote the original comment. It was right at the time, but only in the very limited circumstances of my test systems.

When I realized the true nature of the problem, it also occurred to me that we have to wait until after updateSchema() is called, since it may add tables to the database. Populating the hashes before wouldn't see the new tables.